vcs/displayplot.py

```diff
@@ -116,36 +148,110 @@ class Dp(vcs.bestMatch):
+                            if debug:
+                                with self._parent._display_target.out:
+                                    print("Examing axis:", axId, "vs", ax.axis)
+                            for ax2 in disp.array[1].getAxisList():
+                                if debug:
+                                    with self._parent._display_target.out:
+                                        print("Examing axis:", axId, "vs", ax2.id)
+                                if hasattr(ax2, "axis") and ax2.axis == ax.axis:
+                                    kw2[ax2.id] = kw1[ax.id]
+            if debug:
+                with self._parent._display_target.out:
+                    print("kws", kw1, kw2)
+            if len(kw1) != 0:
+                if debug:
+                    with self._parent._display_target.out:
+                        print("updating :", disp.array[0].id, kw1)
+                new1 = disp.array[0](**kw1)
+                if debug:
+                    with self._parent._display_target.out:
+                        print("updated :", new1.shape)
+            else:
+                new1 = disp.array[0]
+            if len(kw2) != 0:
+                if debug:
+                    with self._parent._display_target.out:
+                        print("updating 2:", kw2)
+                new2 = disp.array[1](**kw2)
+                if debug:
+                    with self._parent._display_target.out:
+                        print("updated 2:", new2.shape)
+            else:
+                if debug:
+                    with self._parent._display_target.out:
+                        print("not updating 2:", disp.array[1])
+                new2 = disp.array[1]
+            if kw1 is not {} or kw2 is not {}:
+                if debug:
+                    with self._parent._display_target.out:
+                        print("calling update")
+                disp._parent.backend.update_input(disp.backend, new1, new2)
+        for widget in widgets:
+            slider, label = widget.children
+            sp = slider.description
+            if debug:
+                with self._parent._display_target.out:
+                    print("OPk looking at:", name, slider.description, sp, sp == name)
+            if sp == name:
+                value = label.values[change["new"]]
+                label.value = "{}".format(value)
+                if debug:
+                    with self._parent._display_target.out:
+                        print("Ok new value:", value, label.value)
         tmp = tempfile.mktemp() + ".png"
         self._parent.png(tmp)
         f = open(tmp, "rb")
         st = f.read()
         f.close()
+        self._parent._display_target.clear_output()
+        with self._parent._display_target:
+            if debug:
+                IPython.display.display(self._parent._display_target.out)
+            IPython.display.display(*widgets)
+            IPython.display.display(IPythonDisplay(st))
```

Comment:
Does this mean `vcs` now depends on IPython?  Or maybe it always did, and I just didn't realize it?

Reply:
it's inside a try that starts with import IPython. So no it does not depend on it but takes advantage of it if present.

Reply:
No you're right it's been moved to its own section but only called from inside the try. I'll still use the variable way you suggested to prevent accidental use.